0.9 JS Fundamentals - DOM/index2.html

```diff
@@ -0,0 +1,122 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<script>
+    document.addEventListener('DOMContentLoaded', async () => {
+        const userTableBody = document.getElementById('userTableBody');
+        const searchInput = document.getElementById('searchInput');
+        const loadingDiv = document.getElementById('loading');
+
+        // Fetch users from mock API
+        const fetchUsers = async () => {
+            try {
+                loadingDiv.style.display = 'block';
+
+                const response = await fetch('https://reqres.in/api/users');
+                const data = await response.json();
+                loadingDiv.style.display = 'none';
+
+                return data.data; // Assuming the API response has a 'data' property containing the user list
+            } catch (error) {
+                loadingDiv.style.display = 'none';
+                console.error('Error fetching users:', error);
+                return [];
+            }
+        };
+
+        // Render users in the table
+        const renderUsers = async () => {
+            const users = await fetchUsers();
+            userTableBody.innerHTML = '';
+
+            users.forEach(user => {
+                const row = userTableBody.insertRow();
+                row.innerHTML = `<td>${user.first_name} ${user.last_name}</td><td>${user.email}</td>`;
+            });
+        };
+
+        // Sort the table by name
+        window.sortTable = (column) => {
```

Comment:
instead of window can we use table data reference here?

Reply:
Yes, instead of using **window.sortTable,** you can directly reference the table data element where you want to attach the **sortTable** function. For example, if you have a table element with an id of **userTableData** , you can define the **sortTable** function as follows:

document.getElementById('userTableData').sortTable = (column) => {
    const table = document.getElementById('userTable');
    const rows = Array.from(table.rows).slice(1); // Exclude header row

    rows.sort((a, b) => {
        const nameA = a.cells[column].textContent.trim();
        const nameB = b.cells[column].textContent.trim();

        return nameA.localeCompare(nameB);
    });

    table.tBodies[0].append(...rows);
};

Reply:
why we are using localeCompare() here?

Reply:
localeCompare() is used for string comparison in JavaScript